packages/firestore/src/lite-api/query.ts

```diff
@@ -193,6 +201,92 @@ export function where(
   return new QueryFilterConstraint(field, op, value);
 }
 
+class QueryCompositeFilterConstraint extends QueryConstraint {
+  constructor(
+    readonly type: 'or' | 'and',
+    private readonly _queryConstraints: QueryFilterConstraint[]
+  ) {
+    super();
+  }
+
+  _parse<T>(query: Query<T>): Filter {
+    const parsedFilters = this._queryConstraints
+      .map(queryConstraint => {
+        return queryConstraint._parse(query);
+      })
+      .filter(parsedFilter => parsedFilter.getFilters().length > 0);
+
+    if (parsedFilters.length === 1) {
+      return parsedFilters[0];
+    }
+
+    return CompositeFilter.create(parsedFilters, this.getOperator());
+  }
+
+  _apply<T>(query: Query<T>): Query<T> {
+    const parsedFilter = this._parse(query);
+    if (parsedFilter.getFilters().length === 0) {
+      // Return the existing query if not adding any more filters (e.g. an empty composite filter).
+      return query;
+    }
+    validateNewFilter(query._query, parsedFilter);
+
+    return new Query(
+      query.firestore,
+      query.converter,
+      queryWithAddedFilter(query._query, parsedFilter)
+    );
+  }
+
+  getQueryConstraints(): QueryConstraint[] {
+    return this._queryConstraints;
+  }
+
+  getOperator(): CompositeOperator {
+    return this.type === 'and' ? CompositeOperator.AND : CompositeOperator.OR;
+  }
+}
+
+/**
+ * Creates a {@link QueryConstraint} that performs a logical OR
+ * of all the provided `queryConstraints`
+ *
+ * @param queryConstraints - Optional. The {@link queryConstraints} for OR operation. These must be
+ * created with calls to {@link where}, {@link or}, or {@link and}.
+ * @returns The created {@link QueryConstraint}.
+ */
+export function or(...queryConstraints: QueryConstraint[]): QueryConstraint {
+  // Only support QueryFilterConstraints
+  queryConstraints.forEach(queryConstraint =>
+    validateQueryFilterConstraint('or', queryConstraint)
+  );
+
+  return new QueryCompositeFilterConstraint(
+    CompositeOperator.OR,
+    queryConstraints as QueryFilterConstraint[]
+  );
+}
+
+/**
+ * Creates a {@link QueryConstraint} that performs a logical AND
+ * of all the provided `queryConstraints`
+ *
+ * @param queryConstraints - Optional. The {@link queryConstraints} for AND operation. These must be
+ * created with calls to {@link where}, {@link or}, or {@link and}.
+ * @returns The created {@link QueryConstraint}.
+ */
+export function and(...queryConstraints: QueryConstraint[]): QueryConstraint {
+  // Only support QueryFilterConstraints
+  queryConstraints.forEach(queryConstraint =>
+    validateQueryFilterConstraint('andQuery', queryConstraint)
```

Comment:
did you figure out the deal with `and` vs `andQuery` ?

Reply:
I was either using the tools incorrectly or missing an export somewhere. But I'm now able to use `and` as the design specified.

Reply:
so I'm guessing you'll replace `andQuery` with `and` here?